website/content/docs/concepts/architecture.md

```diff
@@ -213,3 +213,52 @@ The source for this diagram lives here:
 https://docs.google.com/drawings/d/10d1Q_VO0HFtOHftDV7kK6VbZMVI5EpEYHrD-LR7SczE
 -->
 <img src="/img/topology-submit-sequence-diagram.png" style="max-width:140%;!important;" alt="Topology Sequence Diagram"/>
+
+### Topology Submit Description
+
+The following describes in more detail how a topology is submitted.
+
+* Client
+
+    When a topology is submitted using the `heron submit` command, it first executes
+    the `main` of the topology and creates a `.defn` file containing the topology's
+    logical plan. Then, it runs `com.twitter.heron.scheduler.SubmitterMain`, which
+    is responsible for invoking an uploader and a launcher for the topology.
+    The uploader uploads the topology package to the given location, while the
+    launcher registers the topology's logical plan and executor state with
+    the state manager and invokes the main scheduler.
```

Comment:
according to https://github.com/twitter/heron/pull/973#issuecomment-230018371, use "State Manager"